src/System.Text.Primitives/System/Text/Encoding/Utf16/Utf16TextEncoderLE.cs

```diff
@@ -152,6 +152,29 @@ public override bool TryEncode(string text, Span<byte> encodedBytes, out int byt
             return true;
         }
 
+        public override bool TryComputeEncodedBytes(ReadOnlySpan<byte> utf8, out int bytesNeeded)
+        {
+            int characters;
+            var result = Utf8Encoder.TryComputeStringLength(utf8, out characters);
```

Comment:
Yes. It would still work, though, and is arguably more friendly than the out approach. Is the house policy "tuples on internal API only"? If it is: fine